float32 fX
float32 fY
float32 fDepth
bool bIsDetected